From c66900664e5a0ae5a9894777d8e4b0f2300af95a Mon Sep 17 00:00:00 2001
From: Ankit Jain <ankitja@vmware.com>
Date: Thu, 21 Oct 2021 09:27:53 +0000
Subject: [PATCH] realtime: Modify hung_task detection param

- For realtime system, we wanted to capture
  stack trace if a task remain in D-state for
  30 seconds or more for debugging purpose.
- To log infinite number of times, set
  hung_task_warnings to -1

Signed-off-by: Ankit Jain <ankitja@vmware.com>
---
 profiles/realtime/tuned.conf | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/profiles/realtime/tuned.conf b/profiles/realtime/tuned.conf
index 8eed36e..f9c6883 100644
--- a/profiles/realtime/tuned.conf
+++ b/profiles/realtime/tuned.conf
@@ -36,7 +36,8 @@ isolate_managed_irq = ${isolate_managed_irq}
 managed_irq=${f:regex_search_ternary:${isolate_managed_irq}:\b[y,Y,1,t,T]\b:managed_irq,domain,:}
 
 [sysctl]
-kernel.hung_task_timeout_secs = 600
+kernel.hung_task_timeout_secs = 30
+kernel.hung_task_warnings = -1
 kernel.nmi_watchdog = 0
 kernel.sched_rt_runtime_us = -1
 vm.stat_interval = 10
-- 
2.23.1

